mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -501,7 +501,13 @@ trait Params extends Identifiable with Serializable {
    */
   final def getDefault[T](param: Param[T]): Option[T] = {
     shouldOwn(param)
-    defaultParamMap.get(param)
+    try {
+      defaultParamMap.get(param)
+    } catch {
+      case e: NoSuchElementException =>
```

Comment:
We don't have performance issues here. So code readability is more important, e.g.,

~~~scala
defaultParamMap.getOrElse(param, 
  throw new NoSuchElementException(s"No default value for ${param.name} in ${this.getClass}.")
~~~